assets/training/finetune_acft_hf_nlp/src/validate_lora_weights/validate_lora_weights.py

```diff
@@ -0,0 +1,116 @@
+
+logger = get_logger_app(
+    "azureml.acft.contrib.hf.scripts.components.scripts.validate_lora_weights.validate_lora_weights"
+)
+
+GENERATION_CONFIG = {
+    "max_new_tokens": 256,
+    "do_sample": True,
+    "top_p": 0.7,
+    "temperature": 0.8
+}
+
+
+def validate_lora_weights(base_model_path: str, lora_weights_path: str, tokenizer_path: str, test_examples: list):
+    """Load model and make forward pass to validate lora weights."""
+    logger.info("Validating lora model weights")
+
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_path)
+
+    logger.info("Loading base model")
+    base_model = AutoModelForCausalLM.from_pretrained(base_model_path, device_map="auto", torch_dtype=torch.float16)
+    logger.info("Base model loaded")
+
+    logger.info("Loading lora adapters")
+    model = PeftModel.from_pretrained(base_model, lora_weights_path)
+    logger.info("lora model loaded")
+
+    input_ids = tokenizer(test_examples, return_tensors="pt").input_ids.to("cuda")
+
+    try:
+        outputs = model.generate(inputs=input_ids, **GENERATION_CONFIG)
+        predictions_text = tokenizer.batch_decode(outputs)
+    except Exception:
+        raise ACFTValidationException._with_error(
+                AzureMLError.create(
+                    ACFTUserError,
+                    pii_safe_message=(
+                        "LoRA weights validation failed in forward pass, model cannot be inferenced in fp16"
+                    )
+                )
+            )
+
+    logger.info(f"inputs:\n{test_examples}\ngenerated text:\n{predictions_text}")
+    logger.info("Outputs generated successfully, Validation successful")
+
+
+def get_parser():
+    """Get the parser object."""
+    parser = argparse.ArgumentParser(description="Validate lora weights after finetuning")
+
+    parser.add_argument(
+        "--base_model_path",
+        type=str,
+        required=True,
+        help="Base Model path used for validating lora weights",
+    )
+
+    parser.add_argument(
+        "--lora_weights_path",
+        type=str,
+        required=True,
+        help="LoRA weights path",
+    )
+
+    parser.add_argument(
+        "--tokenizer_path",
+        type=str,
+        required=True,
+        help="Tokenizer path",
+    )
+
+    return parser
+
+
+@swallow_all_exceptions(time_delay=5)
+def main():
+    """Validate lora weights after finetuning."""
+    parser = get_parser()
+    args, _ = parser.parse_known_args()
+
+    set_logging_parameters(
+        task_type="TextGeneration",
+        acft_custom_dimensions={
+            LoggingLiterals.PROJECT_NAME: PROJECT_NAME,
+            LoggingLiterals.PROJECT_VERSION_NUMBER: VERSION,
+            LoggingLiterals.COMPONENT_NAME: COMPONENT_NAME
+        },
+        azureml_pkg_denylist_logging_patterns=LOGS_TO_BE_FILTERED_IN_APPINSIGHTS,
+    )
+
+    print("validating lora weights")
+    validate_lora_weights(args.base_model_path, args.lora_weights_path, args.tokenizer_path, ["Hello"])
```

Comment:
looks like this validation just validate inference capability, but not quality: it cannot prevent inferencing garbage  output?